### PR TITLE
Add model zoo server proto

### PR DESCRIPTION
**Pull request by typhoonzero**
Partwork of implementing the model zoo, design: https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/model_zoo.md and https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/contribute_models_new.md

### CODE REVIEW

**Review by Yancey0623 — APPROVED** [review]
LGTM

File: pkg/proto/modelzooserver.proto [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    bool success = 1;
+    // message describe the error details when error occurs.
+    string message = 2;
+}
+
+message ListModelResponse {
+    // A list of model definition or trained model names.
+    repeated string names = 1;
+    // A list of tags of the above list names, the length should be the same as "names".
+    repeated string tags = 2;
```
Comment (wangkuiyi) [inline pkg/proto/modelzooserver.proto:30]: Why is a model named by a name and a tag?

File: pkg/proto/modelzooserver.proto [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+message Empty {}
+
+service ModelZooServer {
+    // Returns the list of public available model definitions.
+    rpc ListModelDefs (Empty) returns (ListModelResponse);
```
Comment (wangkuiyi) [inline pkg/proto/modelzooserver.proto:37]: How is there are thousands or tens of thousands of model definitions on a server that the protobuf message grows over 32KB?

File: pkg/proto/modelzooserver.proto [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    //    hub.docker.com/my_group/model_image_name, or path of the image to use the default registry:
+    //    my_group/model_image_name
+    // 2. For trained model, it's a unique string describes the model, like: my_resnet50_model_imagenet_acc_0.82
+    string name = 1;
+    // tag of the model definition or trained model, usually a vertion tag like: v0.1
+    string tag = 2;
+    // content_tar is the model definition's directory which contains a Dockerfile and
+    // other files that define some models.
+    // or, can be the content of a trained model.
+    bytes content_tar = 3;
```
Comment (wangkuiyi) [inline pkg/proto/modelzooserver.proto:16]: How comes the value `content_tar`? We need an explanation of the releasing process in the design doc.

File: pkg/proto/modelzooserver.proto [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+message Empty {}
+
+service ModelZooServer {
+    // Returns the list of public available model definitions.
+    rpc ListModelDefs (Empty) returns (ListModelResponse);
+    // Returns the list of public available trained models.
+    rpc ListTrainedModels (Empty) returns (ListModelResponse);
+
+    // Release your model definition.
+    rpc ReleaseModelDef (stream ModelRequest) returns (ModelResponse);
```
Comment (wangkuiyi) [inline pkg/proto/modelzooserver.proto:42]: Why is the parameter a `stream`?

File: pkg/proto/modelzooserver.proto [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+    // Drop your model definition.
+    // DropModelDef use a ModelRequest that have no "content_tar"
+    rpc DropModelDef (ModelRequest) returns (ModelResponse);
+
+    // Release your trained model.
+    rpc ReleaseTrainedModel (stream ModelRequest) returns (ModelResponse);
```
Comment (wangkuiyi) [inline pkg/proto/modelzooserver.proto:48]: Why is the parameter a stream?

File: pkg/proto/modelzooserver.proto [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+    // Drop your trained model.
+    rpc DropTrainedModel (ModelRequest) returns (ModelResponse);
```
Comment (wangkuiyi) [inline pkg/proto/modelzooserver.proto:50]: Why trained model and model definition share the same type `ModelRequest` and `ModelResponse`? In my mind, we can train a model definition with different datasets and hyperparameters and get zero, one, or more trained models.  How do we represent the zero-to-many relationship here?

File: pkg/proto/modelzooserver.proto (hunk 1 above)
Comment (typhoonzero) [reply pkg/proto/modelzooserver.proto:30]: Please checkout https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/model_zoo.md#release-a-trained-model, yet I'll add a database schema description in the `model zoo server` section in this doc.

File: pkg/proto/modelzooserver.proto (hunk 3 above)
Comment (typhoonzero) [reply pkg/proto/modelzooserver.proto:16]: Please refer to  https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/design/model_zoo.md#release-a-trained-model

File: pkg/proto/modelzooserver.proto (hunk 4 above)
Comment (typhoonzero) [reply pkg/proto/modelzooserver.proto:42]: Uploading the model definition folder files need a stream rpc

**Pull request merged**